src/Chapter3.hs

```diff
@@ -1006,6 +1141,71 @@ Implement data types and typeclasses, describing such a battle between two
 contestants, and write a function that decides the outcome of a fight!
 -}
 
+data AttackRound = FighterARound | FighterBRound
+
+nextRound :: AttackRound -> AttackRound
+nextRound FighterARound = FighterBRound
+nextRound FighterBRound = FighterARound
+
+data Result = FighterAWin | FighterBWin deriving (Show, Eq)
+
+determineResult :: AttackRound -> Result
+determineResult FighterARound = FighterAWin
+determineResult FighterBRound = FighterBWin
+
+class Fighter a where
+    getHealth :: a -> Int
+    attacked :: a -> Int -> a
+    doAttack :: (Fighter b) => a -> b -> AttackRound -> Result
+
+-- Knight
+data KKnight = MkKKnight
+    { kknightHealth  :: Int
+    , kknightAttack  :: Int
+    , kknightDefence :: Int
+    }
+
+instance Fighter KKnight where
+    getHealth = kknightHealth
+
+    attacked :: KKnight -> Int -> KKnight
+    attacked kknight attack =
+        let realAttack = max 0 (attack - kknightDefence kknight)
+        in kknight { kknightHealth = kknightHealth kknight - realAttack }
+
+    doAttack :: (Fighter b) => KKnight -> b -> AttackRound -> Result
+    doAttack kknight fighter attackRound =
```

Comment:
Does this function actually need the information about the current Round? I think you can skip this?